manager/apis/app/v1beta1/blueprint_types.go

```diff
@@ -35,6 +36,18 @@ type ApplicationDetails struct {
 	// +optional
 	WorkloadSelector metav1.LabelSelector `json:"selector,omitempty"`
 
+	// Namespaces where user application might run
```

Comment:
I suggest using the entire Selector structure instead of 3 separate fields: WorkloadSelector, Namespaces and IPBlocks.

Reply:
I cannot, this will change backward compatibility